docs/content/guides/upgrading.md

```diff
@@ -1,73 +1,136 @@
+> ```
+
+This will start the Operator with the latest changes, and perform the required sync operations like:
+
+1. Ensuring the CA is still valid
+2. Ensuring a TLS certificate is valid for the local webhook server
+3. If not using the cert-manager integration, patching the Validating and Mutating Webhook Configuration resources with the Capsule CA
+4. If not using the cert-manager integration, patching the Capsule's Custom Resource Definitions conversion webhook fields with the Capsule CA
 
 ## Ensure the conversion webhook is working
 
-Kubernetes Custom Resource definitions provide a conversion webhook that is used by an Operator to perform seamless conversion between resources with different versioning.
+Kubernetes Custom Resource definitions provide a conversion webhook that is used by an Operator to perform a seamless conversion between resources with different versioning.
 
-With the fresh new installation, Capsule patched all the required moving parts to ensure this conversion is put in place, and using the latest version (actually, `v1beta1`) for presenting the Tenant resources.
+With the fresh new installation, Capsule patches all the required moving parts to ensure this conversion is put in place and uses the latest version (actually, `v1beta2`) for presenting the Tenant resources.
 
-You can check this behavior by issuing the following command:
+You can check this behaviour by issuing the following command:
 
 ```
-$: kubectl get tenants.v1beta1.capsule.clastix.io
+$: kubectl get tenants.v1beta2.capsule.clastix.io
 NAME   NAMESPACE QUOTA   NAMESPACE COUNT   OWNER NAME   OWNER KIND   NODE SELECTOR                  AGE
 oil    3                 0                 alice        User         {"kubernetes.io/os":"linux"}   3m43s
 ```
 
 You should see all the previous Tenant resources converted in the new format and structure.
 
 ```
-$: kubectl get tenants.v1beta1.capsule.clastix.io 
+$: kubectl get tenants.v1beta2.capsule.clastix.io 
 NAME   STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR                  AGE
 oil    Active   3                 0                 {"kubernetes.io/os":"linux"}   3m38s
 ```
 
-> Resources are still persisted in etcd using the `v1alpha1` specification and the conversion is executed on-the-fly thanks to the conversion webhook.
+> Resources are still persisted in etcd using the previous Tenant version (`v1beta1`) and the conversion is executed on-the-fly thanks to the conversion webhook.
 > If you'd like to decrease the pressure on Capsule due to the conversion webhook, we suggest performing a resource patching using the command `kubectl replace`:
 > in this way, the API Server will update the etcd key with the specification according to the new versioning, allowing to skip the conversion.
+>
+> The `kubectl replace` command must be triggered when the Capsule webhook is up and running to allow the conversion between versions.
+
+# CapsuleConfiguration new version (v0.2.0)
```

Comment:
Why `CapsuleConfiguration` upgrade is made separately from the controller upgrade? May be I'm missing something.

Reply:
It's stated in the documentation:

> Essentially, the `CapsuleConfiguration` is storing configuration flags that allow Capsule to be configured on the fly without requiring the operator to reload.
> This resource is read at the operator init-time, when the conversion webhook offered by Capsule is not yet ready to serve any request.

During the upgrade phase, no Capsule pods, thus, no conversion webhook, and the CapsuleConfiguration cannot be converted from `v1alpha1` to `v1beta2`.